src/exposure.jl

```diff
@@ -0,0 +1,161 @@
+    "Timestep used in formulation"
+    timestep::TS
+    "The fractional number of rule timesteps in the current simulation timestep"
+    nsteps::S
+end
+Rotation_Exposure{R,W}(;
+    crop=CROP_OCCUPANCY,
+    rotationSize=ROTATION_SIZE,
+    rotationNumber=ROTATION_NUMBER,
+    pulseLevel=PULSE_LEVEL,
+    popThreshold=POPULATION_THRESHOLD,
+    degradationRate=DEGRADATION_RATE,
+    timestep=nothing,
+    nsteps=1.0,
+) where {R,W} = Rotation_Exposure{R,W}(crop, rotationSize, rotationNumber, pulseLevel, popThreshold, degradationRate, timestep, nsteps)
+
+precalcrule(rule::Rotation_Exposure, data) = precalc_timestep(rule, data)
+
+@inline function applyrule(data, rule::Rotation_Exposure, (pesticide, population, rotation), I)
+    crop = get(data, rule.crop, I...) 
+    crop > zero(crop) || return (zero(pesticide), rotation)
+    # return pesticide and rotation
+    # println("rule.rotationNumber $(rule.rotationNumber) == rotation $(rotation.rotationStep)")
+    # println("rotation.rotationStep $(rotation.timeStep) == currentframe $(currentframe(data))")
+    # println(
+    #     $(population >= rule.popThreshold),
+    #     $(mod(rotation.rotationStep, rule.rotationSize) == mod(rule.rotationNumber, rule.rotationSize)),
+    #     $(rotation.timeStep != currentframe(data))")
+    if  ( population >= rule.popThreshold &&
+        mod(rotation.rotationStep, rule.rotationSize) == mod(rule.rotationNumber, rule.rotationSize) &&
+        rotation.timeStep != currentframe(data) )
+        # println("IN $I")
+        @fastmath (
+            pesticide * exp(- rule.degradationRate * rule.nsteps) + rule.pulseLevel,
+            updateStep(rotation + 1, currentframe(data))
+        )
+    else
+        # println("OUT $I")
+        @fastmath (
+            pesticide * exp(- rule.degradationRate * rule.nsteps),
+            rotation
+        )
+    end
+end
+
+struct RotationStruct{RS,TS}
+    rotationStep::RS
+    timeStep::TS
+end
+
+# All numerical operation are on the first element
+Base.:*(rs::RotationStruct, x::Number) = RotationStruct(x * rs.rotationStep, rs.timeStep) 
+Base.:*(x::Number, rs::RotationStruct) = RotationStruct(x * rs.rotationStep, rs.timeStep)
+Base.:+(rs::RotationStruct, x::Number) = RotationStruct(x + rs.rotationStep, rs.timeStep) 
+Base.:+(x::Number, rs::RotationStruct) = RotationStruct(x + rs.rotationStep, rs.timeStep)
+Base.:+(rs1::RotationStruct, rs2::RotationStruct) = RotationStruct(rs1.rotationStep + rs2.rotationStep, rs1.timeStep + rs2.timeStep)
+Base.:-(rs::RotationStruct, x::Number) = RotationStruct(rs.rotationStep - x, rs.timeStep) 
+Base.:-(x::Number, rs::RotationStruct) = RotationStruct(x - rs.rotationStep, rs.timeStep)
+Base.:-(rs1::RotationStruct, rs2::RotationStruct) = RotationStruct(rs1.rotationStep - rs2.rotationStep, rs1.timeStep - rs2.timeStep)
+
+Base.mod(rs::RotationStruct, x::Number) = mod(rs.rotationStep, x)
+
+isEqualStep(rs::RotationStruct, y::Number) = rs.timeStep == y ? true : false
```

Comment:
`isequalstep` or `isequal_step` or something if you really need an underscore. No capitals in function names